### DISCUSSION TITLE
API structure modification

### DISCUSSION THREAD

**Discussion by martinkramerLS** [cloudbet/docs#98] [General]
`"markets": {
                        "soccer.asian_handicap": {
                            "submarkets": {
                                "period=ft": {
                                    "sequence": "132",
                                    "selections": [
                                        {
                                            "outcome": "home",
                                            "params": "handicap=-0.5",
                                            "price": 2.65,
                                            "minStake": 0.1,
                                            "maxStake": 87.30869,
                                            "probability": 0.346,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        },
                                        {
                                            "outcome": "away",
                                            "params": "handicap=-0.5",
                                            "price": 1.4,
                                            "minStake": 0.1,
                                            "maxStake": 360.15,
                                            "probability": 0.654,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        },
                                        {
                                            "outcome": "home",
                                            "params": "handicap=-1.5",
                                            "price": 7,
                                            "minStake": 0.1,
                                            "maxStake": 24.01,
                                            "probability": 0.132,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        },
                                        {
                                            "outcome": "away",
                                            "params": "handicap=-1.5",
                                            "price": 1.06,
                                            "minStake": 0.1,
                                            "maxStake": 2401,
                                            "probability": 0.868,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        },
                                        {
                                            "outcome": "home",
                                            "params": "handicap=0.5",
                                            "price": 1.14,
                                            "minStake": 0.1,
                                            "maxStake": 1029,
                                            "probability": 0.808,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        },
                                        {
                                            "outcome": "away",
                                            "params": "handicap=0.5",
                                            "price": 4.8,
                                            "minStake": 0.1,
                                            "maxStake": 37.91032,
                                            "probability": 0.192,
                                            "status": "SELECTION_ENABLED",
                                            "side": "BACK"
                                        }
                                    ]
                                }
                            }
                        },
                    }`
The structure we've found.
-----
`{
  "market": "Asian Handicap",
  "period": "ft",
  "handicap": "-0,5",
  "selections": [
    {
      "outcome": "home",
      "price": 2.65,
      "minStake": 0.1,
      "maxStake": 87.30869,
      "probability": 0.346,
      "status": "SELECTION_ENABLED",
      "side": "BACK"
    },
    {
      "outcome": "away",
      "price": 1.4,
      "minStake": 0.1,
      "maxStake": 360.15,
      "probability": 0.654,
      "status": "SELECTION_ENABLED",
      "side": "BACK"
    }
  ]
}`
And this would be perfect to have. To have the market name in a "value" and not in an "attribute / key"

**Comment by kgravenreuth** [comment]
Hi @martinkramerLS 

We would suggest you to transform the market data json from our format to the one you desire (i.e. market name as a key value pair of `market: <market name>`) by adding logic to your client to parse and map our input JSON data. 

We provide a list of all our markets, as well as market names in this [Github gist](https://gist.github.com/kgravenreuth/6703e1e213aecac4d5728f2f699d34e7#file-markets-json). So you could add a parsing step to read the input json, and map the market key to market value you desire. 

For example for `soccer.asian_handicap` you can refer to the [market mapping](https://gist.github.com/kgravenreuth/6703e1e213aecac4d5728f2f699d34e7#file-markets-json-L5806-L5807):

```json
	"soccer.asian_handicap": {
		"Name": "Asian Handicap",
```

Hope this helps!

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet
